src/constants/numerical.js

```diff
@@ -36,5 +36,11 @@ module.exports = {
     ONEDAY: 86400000,
     ONEHOUR: 3600000,
     ONEMIN: 60000,
-    ONESEC: 1000
+    ONESEC: 1000,
+
+    /*
+     * For fast conversion btwn world calendars and epoch ms, the Julian Day Number
+     * of the unix epoch. From calendars.instance().newDate(1970, 1, 1).toJD()
+     */
+    EPOCHJD: 2440587.5
```

Comment:
I'm not sure I want to go through with this.

Moving all the logic from the `if(calendar)` blocks to the calendars component would involved passing oddly named variables to new routines and return sets of values and even mutating them in [`findExactDates`](https://github.com/plotly/plotly.js/blob/68af28745fba7551e57ae166a95e64bad76bf6d0/src/lib/dates.js#L555-L567).

I understand that if would be nice to put all the world-calendar logic _in one place_, but it would make the code less readable, unless we start duplicating some `dates.js` logic in the calendars component.

The current implementation is both DRY and readable - which gets my vote.